azurerm/internal/services/databoxedge/databox_edge_orders_resource.go

```diff
@@ -0,0 +1,586 @@
+package databoxedge
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/databoxedge/mgmt/2019-08-01/databoxedge"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/databoxedge/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/databoxedge/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceDataboxEdgeOrder() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDataboxEdgeOrderCreateUpdate,
+		Read:   resourceDataboxEdgeOrderRead,
+		Update: resourceDataboxEdgeOrderCreateUpdate,
+		Delete: resourceDataboxEdgeOrderDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.DataboxEdgeOrderID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"device_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DataboxEdgeName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"contact_information": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"company_name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.DataboxEdgeCompanyName,
+						},
+
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.DataboxEdgeContactName,
+						},
+
+						"emails": {
+							Type:     schema.TypeSet,
+							Required: true,
+							ForceNew: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+
+						"phone_number": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.DataboxEdgePhoneNumber,
+						},
+					},
+				},
+			},
+
+			"current_status": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"status": {
```

Comment:
but then i guess this would need to change 🤔

Reply:
Yeah, since this is inside `current_status` I this is OK.

Reply:
then maybe ther eis an argument ot move these out of this "current_status" block as i don't see the value - order details would make sense without being in a "current status" block

Reply:
I feel the make sense in a block, I think we just need to work the names a bit so they fit 🙂

Reply:
re-worked it a bit, see if you like it any better now... 😁